### PR TITLE
3pcd mitigations post

### PR DESCRIPTION
**Pull request by edunham**
This blog post has the following features:

- [ ] A GitHub Repository with a polished README
- [ ] A GitHub Repository under the github.com/oktadev account
- [ ] A title that's approved by Dev Advocacy
- [ ] A URL approved by Dev Advocacy
- [ ] The content has been run through Grammarly (https://www.grammarly.com/)
- [ ] Rendered locally and confirmed that no Markdown typos exist
- [ ] Images are compressed appropriately
- [ ] Social image previews well on Twitter and LinkedIn
- [ ] Tech review request (developer advocate or domain expert)
- [ ] Review request for editorial/grammar/clarity (developer advocate or Colton)

### CODE REVIEW

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+You may be able to use this solution if your apps are hosted on your domain. Usually the URL for those apps starts with something like app.yourdomain.com.
+
+Usually the session cookie is issued from your Okta domain host, from yourdomain.okta.com. Because of that it looks like a third-party cookie to your own domain. To make it a first party cookie, you can configure a custom domain in the Okta admin console so that it's issued from `okta.yourdomain.com`.
+
+The custom domain for your Okta organization should be a subdomain of your own domain. If you don't use network zones, you can use an Okta-managed [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) certificate for a custom domain. If you do use network zones, you'll need to provide your own TLS cert by following [Okta's instructions](https://developer.okta.com/docs/guides/custom-url-domain/main/#use-your-own-tls-certificate).
+
+#### To configure your custom domain:
+
+1) Open the Admin console for your Okta org.
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
1. Open the Admin console for your Okta org.
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+2) In the Admin Console, select **Customizations**.
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
2. In the Admin Console, select **Customizations**.
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+3) Select **Domain** in the customization list.
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
3. Select **Domain** in the customization list.
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+4) In the **Custom URL Domain** section, click **Edit** and follow the instructions on the page that appears.
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
4. In the **Custom URL Domain** section, click **Edit** and follow the instructions on the page that appears.
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+If you have customized the look and feel of the Okta sign-in page using resources like images that are hosted on other domains, you will need to configure the Content Security Policy (CSP) to allow those resources. Learn more about customizing your CSP in [this Okta help center guide](https://help.okta.com/oie/en-us/content/topics/settings/customizations-configure-csp.htm). 
+
+### Use the authorization code flow
+
+Changing your application to use [the authorization code flow](https://developer.okta.com/docs/guides/implement-grant-type/authcode/main/) ensures that your sign-in widget functions in browsers with third-party cookies disabled. 
+
+#### To change an Okta application to use the authorization code flow:
+
+1) Open the Admin console for your Okta org.
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
1. Open the Admin console for your Okta org.
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+2) In the Admin Console, select Applications > Applications.
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
2. In the Admin Console, select Applications > Applications.
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+3) Open the desired app integration.
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
3. Open the desired app integration.
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+4) Select General.
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
4. Select General.
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+5) In the General Settings section, select the Refresh Token grant type and configure the desired token rotation policy. "Rotate token after every use" is the most secure option and is suitable for most cases.
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
5. In the General Settings section, select the Refresh Token grant type and configure the desired token rotation policy. "Rotate token after every use" is the most secure option and is suitable for most cases.
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+With these flags set, sign into Okta with your self-hosted sign-in widget. Check the Chrome developer console to see if your code is raising any errors, and verify that you're not prompted to log in again more often than you expect. 
+
+### Application behaviors to look for
+
+Cookies are used to introspect and extend the Okta session. If a third-party cookie was previously being used to introspect the session and gets blocked, you'll probably see errors in the developer console. If a third-party cookie was used to extend the session, when it's blocked you'll be logged out at session extension time. Note that this may occur some time after the login.
+
+## Temporary fix in 2024
+
+Google granted Okta an exemption to the third-party cookie deprecation rollout until the end of 2024. To opt in to this exemption, you embed an Okta-provided script into your sign-in widget following the instructions at https://support.okta.com/help/s/article/third-party-cookies-utilized-by-the-sign-in-widget?language=en_US. The script sets a Trial token, which tells Google to continue allowing third-party cookies for your login experience until the end of 2024.
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
Google granted Okta an exemption to the third-party cookie deprecation rollout until the end of 2024. To opt in to this exemption, you embed an Okta-provided script into your sign-in widget following [these instructions](https://support.okta.com/help/s/article/third-party-cookies-utilized-by-the-sign-in-widget?language=en_US). The script sets a Trial token, which tells Google to continue allowing third-party cookies for your login experience until the end of 2024. 
```

**Review by edunham — COMMENTED** [review]
fix formatting

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+#### To change an Okta application to use the authorization code flow:
+
+1) Open the Admin console for your Okta org.
+2) In the Admin Console, select Applications > Applications.
+3) Open the desired app integration.
+4) Select General.
+5) In the General Settings section, select the Refresh Token grant type and configure the desired token rotation policy. "Rotate token after every use" is the most secure option and is suitable for most cases.
+
+For more information on the authorization code flow, see (Get a refresh token with the code flow in Refresh access tokens)[https://developer.okta.com/docs/guides/refresh-tokens/main/#get-a-refresh-token-with-the-code-flow].
```
Comment (isemona) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]: brackets on the title instead for the hyperlink
```suggestion
For more information on the authorization code flow, see [Get a refresh token with the code flow in Refresh access tokens](https://developer.okta.com/docs/guides/refresh-tokens/main/#get-a-refresh-token-with-the-code-flow).
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+Just like other scopes, the `offline_access` scope must be enabled in both the Okta admin console and in your application code. Once you set the `offline_access` scope on your Authorization Server and in your AuthJS code, AuthJS uses refresh tokens instead of cookies. 
+
+To set the `offline_access` scope on the Authorization Server: 
+
+1) Open the Admin console for your Okta org.
+2) In the side menu of the Admin Console, navigate to **Security** and select **API**.
+3) In the **Authorization Servers** tab, click the pencil-shaped Edit icon to the right of the authorization server that your application uses.
+4) Navigate to the **Scopes** tab of the authorization server's page
+5) Use the **Add Scope** button to add the `offline_access` scope.
+
```
Comment (isemona) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md:78]: The above is not showing up as a list in deploy preview for some reason.

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+author: edunham
+by: advocate
+communities: [devops,security,mobile,.net,java,javascript,go,php,python,ruby]
+description: "If you have a self-hosted Okta Sign-In Widget, there are several options to mitigate Chrome's third-party cookies deprecation."
+tags: [third-party-cookies, okta-sign-in-widget]
+tweets:
+- ""
+- ""
+- ""
+image: blog/3pc/social.jpg
```
Comment (isemona) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]: blog image is the same as the 2/29 post?

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+
+1) Open the Admin console for your Okta org.
+2) In the side menu of the Admin Console, navigate to **Security** and select **API**.
+3) In the **Authorization Servers** tab, click the pencil-shaped Edit icon to the right of the authorization server that your application uses.
+4) Navigate to the **Scopes** tab of the authorization server's page
+5) Use the **Add Scope** button to add the `offline_access` scope.
+
+To set the `offline_access` scope in your AuthJS code, add `offline_access` to the `scopes` array in your `OktaAuth` configuration. This configuration is found in the file where you [initialize the SDK](https://developer.okta.com/docs/guides/auth-js/main/#initialize-the-sdk).
+
+
```
Comment (isemona) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md:84]:
```suggestion
> **Note**
>
> This to edit a [custom authorization server](https://developer.okta.com/docs/concepts/auth-servers/#custom-authorization-server) requires the feature [API Access Management](https://www.okta.com/products/api-access-management/?adgroupid=&campaignid=&utm_source=google&utm_campaign=amer_mult_usa_all_wf-all_dg-ao_a-wf_search_google_text_kw_dsa_utm2&utm_medium=cpc&utm_id=aNK4z000000UAzJGAW&gad_source=1&gclid=Cj0KCQjwztOwBhD7ARIsAPDKnkD3WQMla4xEM7GwHXqaQt-O2IF6mH1x5dadKgpUp4rH12IYyZQgjP4aApevEALw_wcB) enabled.
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 15]
```diff
@@ -0,0 +1,113 @@
+---
+layout: blog_post
+title: "How to prepare your self-hosted Okta Sign-In Widget to work without third-party cookies"
+author: edunham
+by: advocate
+communities: [devops,security,mobile,.net,java,javascript,go,php,python,ruby]
+description: "If you have a self-hosted Okta Sign-In Widget, there are several options to mitigate Chrome's third-party cookies deprecation."
+tags: [third-party-cookies, okta-sign-in-widget]
```
Comment (isemona) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
tags: [third-party-cookies, okta-sign-in-widget, auth-js-sdk]
```

**Review by isemona — COMMENTED** [review]
Lastly, maybe a CTA to comment below for questions?

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md (hunk 13 above)
Comment (edunham) [reply _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]: yeah thanks to Security(TM) I don't actually have access to appropriate software for making a new one right now :/

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 16]
```diff
@@ -0,0 +1,115 @@
+---
+layout: blog_post
+title: "How to prepare your self-hosted Okta Sign-In Widget to work without third-party cookies"
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
title: "How to Prepare Your Self-Hosted Okta Sign-in Widget to Work without Third-Party Cookies"
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md (hunk 12 above)
Comment (edunham) [reply _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md:78]: fixed, i forgot how to markdown for a sec there

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+
+1) Open the Admin console for your Okta org.
+2) In the side menu of the Admin Console, navigate to **Security** and select **API**.
+3) In the **Authorization Servers** tab, click the pencil-shaped Edit icon to the right of the authorization server that your application uses.
+4) Navigate to the **Scopes** tab of the authorization server's page
+5) Use the **Add Scope** button to add the `offline_access` scope.
+
+To set the `offline_access` scope in your AuthJS code, add `offline_access` to the `scopes` array in your `OktaAuth` configuration. This configuration is found in the file where you [initialize the SDK](https://developer.okta.com/docs/guides/auth-js/main/#initialize-the-sdk).
+
+> **Note**
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md:81]:
```suggestion
**Note**
```

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+>
+> This to edit a [custom authorization server](https://developer.okta.com/docs/concepts/auth-servers/#custom-authorization-server) requires the feature [API Access Management](https://www.okta.com/products/api-access-management/?adgroupid=&campaignid=&utm_source=google&utm_campaign=amer_mult_usa_all_wf-all_dg-ao_a-wf_search_google_text_kw_dsa_utm2&utm_medium=cpc&utm_id=aNK4z000000UAzJGAW&gad_source=1&gclid=Cj0KCQjwztOwBhD7ARIsAPDKnkD3WQMla4xEM7GwHXqaQt-O2IF6mH1x5dadKgpUp4rH12IYyZQgjP4aApevEALw_wcB) enabled.
+## Transition to supported endpoints
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md:85]:
```suggestion

## Transition to supported endpoints
```

**Review by isemona — APPROVED** [review]
👍

File: _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+2) In the side menu of the Admin Console, navigate to **Security** and select **API**.
+3) In the **Authorization Servers** tab, click the pencil-shaped Edit icon to the right of the authorization server that your application uses.
+4) Navigate to the **Scopes** tab of the authorization server's page
+5) Use the **Add Scope** button to add the `offline_access` scope.
+
+To set the `offline_access` scope in your AuthJS code, add `offline_access` to the `scopes` array in your `OktaAuth` configuration. This configuration is found in the file where you [initialize the SDK](https://developer.okta.com/docs/guides/auth-js/main/#initialize-the-sdk).
+
+> **Note**
+>
+> This to edit a [custom authorization server](https://developer.okta.com/docs/concepts/auth-servers/#custom-authorization-server) requires the feature [API Access Management](https://www.okta.com/products/api-access-management/?adgroupid=&campaignid=&utm_source=google&utm_campaign=amer_mult_usa_all_wf-all_dg-ao_a-wf_search_google_text_kw_dsa_utm2&utm_medium=cpc&utm_id=aNK4z000000UAzJGAW&gad_source=1&gclid=Cj0KCQjwztOwBhD7ARIsAPDKnkD3WQMla4xEM7GwHXqaQt-O2IF6mH1x5dadKgpUp4rH12IYyZQgjP4aApevEALw_wcB) enabled.
```
Comment (edunham) [inline _source/_posts/2024-04-08-third-party-cookies-okta-sign-in-widget.md]:
```suggestion
> This is to edit a [custom authorization server](https://developer.okta.com/docs/concepts/auth-servers/#custom-authorization-server) and requires that the feature [API Access Management](https://www.okta.com/products/api-access-management/?adgroupid=&campaignid=&utm_source=google&utm_campaign=amer_mult_usa_all_wf-all_dg-ao_a-wf_search_google_text_kw_dsa_utm2&utm_medium=cpc&utm_id=aNK4z000000UAzJGAW&gad_source=1&gclid=Cj0KCQjwztOwBhD7ARIsAPDKnkD3WQMla4xEM7GwHXqaQt-O2IF6mH1x5dadKgpUp4rH12IYyZQgjP4aApevEALw_wcB) is enabled for the okta org.
```

**Pull request merged**